assets/src/adaptivity/operators/equality.ts

```diff
@@ -51,6 +51,9 @@ export const isEqual = (factValue: any, value: any): boolean => {
     return value.toString().toLowerCase() === 'true' ? true === factValue : false === factValue;
   }
   if (typeOfValue === 'boolean') {
+    if (typeOfFactValue === 'number') {
+      return factValue === 1 ? true : false;
```

Comment:
I think with visits it can be more than 1, so probably you want to check `>0` for truthiness of numbers. also you don't need to do a tertiary for boolean, just `return factValue > 0;`

Reply:
oh and also we should add unit tests for these new cases

Reply:
Unit test added and refactored the code to match all the use cases